doc/changelog.rst

```diff
@@ -6,6 +6,7 @@ Changelog
 
 Breaking changes:
 
+* Azure Blob Storage SDK has been upgraded from the legacy version (v2) to the latest version (v12). AzureFileAdmin now accept `blob_service_client` rather than `connection_string` to give more flexibility with connection types.
```

Comment:
Tiny one: this is now in the wrong block because the main branch has changed under this PR. If you're happy to rebase/merge in master, that'd be helpful. Otherwise I can tidy this up after the fact.